tests/unit/lib/test_hf.py

```diff
@@ -50,26 +54,28 @@ def test_hf_class_label():
 
 def test_hf_sequence_list():
     ds = Dataset.from_dict({"seq": [[0, 1], [2, 3]]})
-    schema = get_output_schema(ds.features)
+    schema, norm_names = get_output_schema(ds.features)
     assert schema["seq"] == list[int]
 
-    gen = HFGenerator(ds, dict_to_data_model("", schema))
+    gen = HFGenerator(ds, dict_to_data_model("", schema, list(norm_names.values())))
     gen.setup()
     row = next(iter(gen.process()))
     assert row.seq == [0, 1]
 
 
 def test_hf_sequence_dict():
+    # ? in the column name should be normalized
+    # Check if even nested names are not normalized we handle it correctly
     ds = Dataset.from_dict(
-        {"pokemon": [{"name": ["bulbasaur"]}, {"name": ["squirtle"]}]}
+        {"pokemon": [{"name?": ["bulbasaur"]}, {"name?": ["squirtle"]}]}
```

Comment:
Do we also need to check double underscore in tests? Something like `na__me` should also use alias?